### ISSUE TITLE
Induce rehash after pip install cmake

### ISSUE THREAD

**Issue by NAThompson** [scikit-build/cmake-python-distributions#295]
Into a clean virtualenv, I can:

```
pip install cmake
```

Immediately after, I run:

```
$ cmake ../ -G Ninja
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.20.2 or higher is required.  You are running version 3.5.2


-- Configuring incomplete, errors occurred!
```

The workaround is simple: Run a `rehash` immediately after `pip install cmake`. However, I wonder if this can be achieved automatically via some pip hook.

(BTW: Amazing and useful package. Thanks so much!)

**Comment by henryiii** [comment]
Nope, wheels are not allowed to run custom code on install. Partially security issue. The previous format (eggs) did.

**Comment by henryiii** [comment]
Does this happen if you use pipx? I'm not using a shell where rehashes are needed.
